### PR TITLE
Modify project to remove KernelAbstractions, add CUDA to deps, remove StaticArrays

### PR DESCRIPTION
**Pull request by akshaysridhar**
Towards GPU compatibility

Removes KernelAbstractions and CUDAKernels from Project.toml
Removes StaticArrays from deps.
Adds CUDA to Project.
Adds compat entries for relevant packages.
Applies same changes to test environment.

### CODE REVIEW

File: src/SurfaceFluxes.jl [hunk 1]
```diff
@@ -416,31 +415,31 @@ function obukhov_length(
...
-                KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
-                KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
-                KA.@print(", Δz = ", Δz(sc))
-                KA.@print(", ΔDSEᵥ = ", ΔDSEᵥ)
-                KA.@print("\n")
-                KA.@print("ts_in(sc) = ", ts_in(sc))
-                KA.@print("\n")
-                KA.@print("ts_sfc(sc) = ", ts_sfc(sc))
-                KA.@print("\n")
+                @printf("maxiter reached in SurfaceFluxes.jl:\n")
```
Comment (charleskawczynski) [inline src/SurfaceFluxes.jl:418]: The KernelAbstractions `@print` is GPU compatible. And my guess is that`@printf` is not

Comment (akshaysridhar) [reply src/SurfaceFluxes.jl:418]: Right - for now this PR switches from KA -> CUDA + ClimaComms, the idea was to re-introduce the GPU print in the next PR. Would [CUDA.@cuprintf](https://docs.juliahub.com/CUDA/oWw5k/1.2.1/api/kernel/#CUDA.@cuprintf) this be a reasonable alternative ?

Comment (charleskawczynski) [reply src/SurfaceFluxes.jl:418]: Maybe, but the KernelAbstractions version adapts between CPU and GPU

**Comment by akshaysridhar** [comment]
Closing - the changes suggested in this PR have been broken up into smaller pieces.

**Pull request closed without merging**